server/models/Application.ts

```diff
@@ -78,37 +123,13 @@ const Application = sequelize.define(
     },
   },
   {
+    sequelize,
+    modelName: 'Application',
     paranoid: true,
-
-    getterMethods: {
-      info() {
-        return {
-          name: this.name,
-          description: this.description,
-          apiKey: this.apiKey,
-          clientId: this.clientId,
-          clientSecret: this.clientSecret,
-          callbackUrl: this.callbackUrl,
-        };
-      },
-      publicInfo() {
-        return {
-          id: this.id,
-          name: this.name,
-          description: this.description,
-          type: this.type,
-          CreatedByUserId: this.CreatedByUserId,
-          CollectiveId: this.CollectiveId,
-          createdAt: this.createdAt,
-          updatedAt: this.updatedAt,
-          deletedAt: this.deletedAt,
-        };
-      },
-    },
   },
 );
 
-Application.create = props => {
+Application.create = (props?): Promise<ReturnType<typeof Application.create>> => {
```

Comment:
Maybe simpler

```suggestion
Application.create = (props?): Promise<Application> => {
```

Reply:
`Promise<Application>` doesn't work as the return type. `tsc` gives this error
```
server/models/Application.ts:132:1 - error TS2322: Type '<M extends Model<any, any>, O extends CreateOptions<Attributes<M>> = CreateOptions<Attributes<M>>>(this: ModelStatic<...>, props?: MakeNullishOptional<M["_creationAttributes"]>) => Promise<...>' is not assignable to type '<M extends Model<any, any>, O extends CreateOptions<Attributes<M>> = CreateOptions<Attributes<M>>>(this: ModelStatic<...>, values?: MakeNullishOptional<M["_creationAttributes"]>, options?: O) => Promise<...>'.
  Type 'Promise<Application>' is not assignable to type 'Promise<O extends { returning: false; } | { ignoreDuplicates: true; } ? void : M>'.
    Type 'Application' is not assignable to type 'O extends { returning: false; } | { ignoreDuplicates: true; } ? void : M'.

132 Application.create = (props?): Promise<Application> => {
    ~~~~~~~~~~~~~~~~~~


Found 1 error in server/models/Application.ts:132

```

Reply:
Then we need to understand why and fix the root cause, but there's no double that `Application.create` should return an `Application`

Reply:
The problem here is that `tsc` expects the function assigned to `Application.create` to have the same return type (`Promise<Model>`) as `Model.create`  method - since we are inheriting from `Model`.

We are still returning `Application` from `Application.create`.

Reply:
You can safely ignore `CI / GraphQL Inspector (pull_request)`, it is not working on forks